src/IO.h

```diff
@@ -156,6 +180,8 @@ void WritePredictionsToFiles(Status status, std::vector<std::vector<int>>& end_c
 template <typename ParseVariableType>
 ParseVariableType ReadNumberFromFile(std::string SettingName, std::string SettingCategory, const std::string& filePath) 
 {
+	if (!Utilities::fileExists(filePath)) { throw IOException(filePath + ": File not found!"); }
+
 	std::string SettingValue = CIniFile::GetValue(SettingName, SettingCategory, filePath);
```

Comment:
I think there should be an extra check here. If the value was not found in the INI file, an empty string is returned by `GetValue`. If this happens, an exception should be thrown, explicitly stating that the value was not found in the file.